### PR TITLE
Incremental migration #129

### PR DESCRIPTION
**Pull request by vackosar**

### CODE REVIEW

File: spline-shadow/migrator-tool/src/test/scala/za/co/absa/spline/migrator/MongoStreamMigratorSpec.scala [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+import za.co.absa.spline.model.dt.Simple
+import za.co.absa.spline.model.op.{BatchWrite, Generic, OperationProps}
+import za.co.absa.spline.model.{Attribute, DataLineage, MetaDataset, Schema}
+import za.co.absa.spline.persistence.mongo.dao.{LineageDAOv3, LineageDAOv4, LineageDAOv5, MultiVersionLineageDAO}
+import za.co.absa.spline.persistence.mongo.{MongoConnectionImpl, MongoDataLineageWriter}
+import za.co.absa.spline.persistence.{ArangoFactory, ArangoInit}
+
+import scala.concurrent.{ExecutionContext, Future}
+
+class MongoStreamMigratorSpec extends AsyncFunSpec with Matchers with BeforeAndAfterEach {
```
Comment (wajda) [inline spline-shadow/migrator-tool/src/test/scala/za/co/absa/spline/migrator/MongoStreamMigratorSpec.scala]: Should be integration test, not unit test.

File: parent/pom.xml [hunk 2]
```diff
@@ -387,6 +387,12 @@
                 <groupId>com.github.salat</groupId>
                 <artifactId>salat-core_${scala.compat.version}</artifactId>
                 <version>${salat.version}</version>
+                <exclusions>
```
[the remark this answers is not in the document]
Comment (vackosar) [reply parent/pom.xml]: Salat depends an old version. But it is rather left over from an experiment.

File: spline-shadow/migrator-tool/src/main/scala/za/co/absa/spline/migrator/MigratorCLI.scala [hunk 3]
```diff
@@ -49,9 +49,13 @@ object MigratorCLI extends App {
       text s"Number of batches to process. Negative value means unbounded. (Default is ${MigratorConfig.empty.batchesMax})"
       action ((value, conf) => conf.copy(batchesMax = value)))
 
-    (opt[Unit]('i', "init-arango-db")
+    (opt[Boolean]('i', "init-arango-db")
```
[the remark this answers is not in the document]
Comment (vackosar) [reply spline-shadow/migrator-tool/src/main/scala/za/co/absa/spline/migrator/MigratorCLI.scala]: bit more explicit

[the remark this answers is not in the document]
Comment (vackosar) [reply spline-shadow/migrator-tool/src/main/scala/za/co/absa/spline/migrator/MigratorCLI.scala]: But you are right, I am removing this one.

File: integration-tests/src/test/scala/za/co/absa/spline/MongoStreamMigratorSpec.scala [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+  private def isLineageStoredInArango(lineageId: String) =
+    arangodb
+      .collection("execution")
+      .documentExists(lineage.id.replaceFirst("ln_", ""))
+
+  override protected def beforeEach(): Unit = dropAndInitDbs()
+
+  override protected def afterEach(): Unit = dropAndInitDbs()
+
+  private def dropAndInitDbs(): Unit = {
```
Comment (wajda) [inline integration-tests/src/test/scala/za/co/absa/spline/MongoStreamMigratorSpec.scala]: This behaviour should be a part of the respective fixture. Unlike an ordinary utility object that only provides a bunch of methods but requires a client to call them properly and in the right order, a fixture does everything. It is responsible for  setting up test preconditions and cleaning the used resources up after the test. To do this a fixture is aware of the testing framework and hooks up to the test life cycle accordingly.

Please take a look at the implementation and usage of `SplineFixture` as an example.

**Pull request merged**